rclcpp/src/rclcpp/executor.cpp

```diff
@@ -679,24 +676,13 @@ Executor::get_next_ready_executable(AnyExecutable & any_executable)
   }
 
   if (!valid_executable) {
-    size_t current_timer_index = 0;
-    while (true) {
-      auto [timer, timer_index] = wait_result_->peek_next_ready_timer(current_timer_index);
-      if (nullptr == timer) {
-        break;
-      }
-      current_timer_index = timer_index;
+    while (auto timer = wait_result_->next_ready_timer()) {
       auto entity_iter = current_collection_.timers.find(timer->get_timer_handle().get());
       if (entity_iter != current_collection_.timers.end()) {
         auto callback_group = entity_iter->second.callback_group.lock();
-        if (callback_group && !callback_group->can_be_taken_from()) {
+        if (!callback_group || !callback_group->can_be_taken_from()) {
```

Comment:
Does this mean this pr supersedes https://github.com/ros2/rclcpp/pull/2476?

Reply:
I cherry picked the test case, and changed it to snake_case.

Note, this PR is an indirect followup of #2476, as it fixes the test cases that popped.